### PR TITLE
Upgrade kyma from 2.9.3 to 2.11.4

### PR DESCRIPTION
**Pull request by nyordanoff**
**Description**
This PR upgrades the Kyma version from 2.9.3 to 2.11.4

Changes proposed in this pull request:
- bumped kyma and kyma cli versions
- as of version 1.16 of Istio (which comes with this Kyma upgrade), the `sidecar.istio.io/inject` annotation was deprecated in favour of the `sidecar.istio.io/inject` label
- updated docs

**Related issue(s)**

- ...

**Pull Request status**

- [ ] Implementation
- [ ] Unit tests
- [ ] Integration tests
- [ ] `chart/compass/values.yaml` is updated 
- [ ] Mocks are regenerated, using the automated script

### CODE REVIEW

**Comment by nyordanoff** [comment]
Тhe benchmark job is expected to fail because firstly it installs compass from main branch (that creates a certs-setup-job Job) and then when compass is reinstalled from the branch, there is a change that adds new label to this Job’s pod (the job already exists and a patch is performed but the labels field is immutable)

File: docs/compass/04-01-installation.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ Therefore, `serviceAccountTokenJWKS` and `serviceAccountTokenIssuer` need to be
 
 > **NOTE:** During the installation of Compass, the installed Kyma version (as a basis to Compass) must match to the one in the [`KYMA_VERSION`](../../installation/resources/KYMA_VERSION) file in the specific Compass commit.
 
-If custom domains and certificates are needed, see the [Set up your custom domain TLS certificate](https://github.com/kyma-project/kyma/blob/2.9.3/docs/03-tutorials/00-security/sec-01-tls-certificates-security.md) document in the Kyma installation guide, as well as the resources in the [Certificate Management](#certificate-management) section in this document.
+If custom domains and certificates are needed, see the [Set up your custom domain TLS certificate](https://github.com/kyma-project/kyma/blob/2.11.2/docs/03-tutorials/00-security/sec-01-tls-certificates-security.md) document in the Kyma installation guide, as well as the resources in the [Certificate Management](#certificate-management) section in this document.
```
Comment (ognyvrac) [inline docs/compass/04-01-installation.md:49]: Isn't it `2.11.4`? Same for the other docs

File: chart/compass/templates/tests/tenant-fetcher/tenant-fetcher-test.yaml [hunk 2]
```diff
@@ -15,10 +15,10 @@ spec:
   template:
     metadata:
       annotations:
-        sidecar.istio.io/inject: "true"
         cluster-autoscaler.kubernetes.io/safe-to-evict: "false"
       labels:
         app: {{ .Chart.Name }}-tenant-fetcher-tests-app
+        sidecar.istio.io/inject: "true"
```
Comment (ognyvrac) [inline chart/compass/templates/tests/tenant-fetcher/tenant-fetcher-test.yaml:21]: Giving Atlas a heads-up on this (annotation replaced by label) could be a good idea

File: docs/compass/04-01-installation.md (hunk 1 above)
Comment (nyordanoff) [reply docs/compass/04-01-installation.md:49]: Ah, `2.11.2` is the Kyma CLI version and it seems that I misconfigured them in the docs. Will update it, thanks.

**Review by ognyvrac — APPROVED** [review]

**Pull request merged**